modules/core/shared/src/main/scala/gem/GuideOptions.scala

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2016-2018 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem
+
+import gem.util.Zipper
+
+import cats._
+import cats.implicits._
+
+
+/**
+ * Each observation has a collection of guide options, where each option is a
+ * guide group. One of these options is automatically managed by the auto-guide
+ * star service.  The rest, if any, are manually added by the user.  There can
+ * be zero or one selected groups.  The selection indicates which group will be
+ * used for observing.
+ *
+ * @param auto   automatically managed guide group
+ * @param manual manually added guide groups, where `Left` indicates that none
+ *               of the manual groups is the selected group while `Right`
+ *               indicates that a particular manual group (the zipper focus)
+ *               should be used instead of the auto group
+ *
+ * @tparam G guide group type
+ */
+final case class GuideOptions[G](
+  auto:   Option[G],
+  manual: Either[List[G], Zipper[G]]
+) {
+
+  def all: List[G] = {
+    def ms = manual.fold(identity, _.toList)
+    auto.fold(ms)(_ :: ms)
```

Comment:
FYI you can say `auto.foldRight(ms)(_ :: _)` if you want to feel clever.

Reply:
Nice. 👍  That would enable a one-liner implementation:

`auto.fold(manual.map(_.toList).merge)(_ :: _)`

Reply:
Hmm, I ended up needing an explicit conversion to `Iterator` to avoid picking up the cats `Foldable` version with `Eval`.  Short of fine grained import management, is there a better way to do this?

```scala
  def all: List[G] =
    auto.iterator.foldRight(manual.map(_.toList).merge)(_ :: _)
```

Reply:
Ah, yeah. Stdlib has a conversion to `Iterable` and cats has a conversion to `Foldable.Ops` or something and that's the one you get since it's imported explicitly. At this point what you had first is probably easier to understand. :-\